api/services/orders/order_handler.go

```diff
@@ -274,3 +275,24 @@ func isValidStatusUpdate(currentStatus string, newStatus string) error {
 
 	return fmt.Errorf("order status can not change from '%v' to '%v'", currentStatus, newStatus)
 }
+
+// @Summary Retrieve order details by invoice number
+// @Description Retrieves order details and item list by invoice number.
+// @Tags Orders
+// @Produce json
+// @Security Bearer
+// @Param invoice_number path string true "Invoice Number"
+// @Success 200 {object} types.OrderByInvoiceResponse
+// @Router /orders/invoice/{invoice_number} [get]
+func (h *OrderHandler) handleGetOrderByInvoice(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	invoiceNumber := vars["invoice_number"]
+
+	response, err := h.store.GetOrderByInvoice(invoiceNumber)
+	if err != nil {
```

Comment:
I think you need to have two conditions:

1. If `err != nil`: this is an `InternalServerError` returning the `err`
2. If `order == nil`: this a `StatusNotFound` with the message you have

Reply:
Is this better?
<img width="891" alt="image" src="https://github.com/user-attachments/assets/29fdbe0e-d04d-4afe-a4a4-0594b09bc10f">

Reply:
for the `err != nil` case you can just do `WriteError(w, ..., err)`

Reply:
<img width="491" alt="image" src="https://github.com/user-attachments/assets/16b3f368-6e02-49c4-b721-3ef2bb065a3e">